### PR TITLE
Improve rule group metrics

### PR DESCRIPTION
**Pull request by brian-brazil**
This adds per-rule group duration and interval, so you can see where your time is being spent. From there you can go to the UI to find the particular rules which are problems.

This also changes the names of existing metrics and removes labels, to make them easier to understand.

Example output: https://gist.github.com/brian-brazil/81f4e71cbc9c5bd384b8e11c5ee0922f

@brancz

### CODE REVIEW

File: rules/manager.go [hunk 1]
```diff
@@ -41,68 +41,65 @@ import (
...
-		[]string{"rule_type"},
 	)
 	iterationDuration = prometheus.NewSummary(prometheus.SummaryOpts{
 		Namespace:  namespace,
-		Name:       "evaluator_duration_seconds",
+		Name:       "rule_group_duration_seconds",
 		Help:       "The duration of rule group evaluations.",
 		Objectives: map[float64]float64{0.01: 0.001, 0.05: 0.005, 0.5: 0.05, 0.90: 0.01, 0.99: 0.001},
 	})
 	iterationsMissed = prometheus.NewCounter(prometheus.CounterOpts{
```
Comment (Gouthamve) [inline rules/manager.go]: Does it make sense to put `rule_group` as a label here and in `rule_group_iterations_total`

Comment (brian-brazil) [reply rules/manager.go]: If you did that then you wouldn't be able to measure the total time spent by all rule groups, as they could appear and disappear.

Comment (Gouthamve) [reply rules/manager.go]: Sorry, I actually meant for `rule_group_iterations_missed_total` not `rule_group_duration_seconds`

Comment (brian-brazil) [reply rules/manager.go]: Same argument applies, and you'll be able to detect that per rule group by the last eval being higher than the interval.

File: rules/manager.go [hunk 2]
```diff
@@ -625,3 +617,25 @@ func (m *Manager) AlertingRules() []*AlertingRule {
 	}
 	return alerts
 }
+
+// Implements prometheus.Collector.
```
Comment (Gouthamve) [inline rules/manager.go]: From the codacy check, should be `Describe Implements prometheus.Collector` Similarly below.

**Pull request merged**